crates/wasmtime/src/module.rs

```diff
@@ -360,7 +357,12 @@ impl Module {
             .context("failed to parse WebAssembly module")?;
         let types = types.finish();
         let (mmap, info) = Module::compile_functions(engine, translation, &types)?;
-        Ok((mmap, info, types))
+        Ok((
+            mmap,
+            info,
+            types,
+            engine.is_branch_protection_enabled(&engine.compiler().isa_flags()),
```

Comment:
Since this an be inferred from `&Engine`-the-argument could this return value be dropped?

Reply:
This is a bit tricky, but the reason is the cache implementation on line 291, and in particular the call to `SerializedModule::into_parts()` on line 314, which, given the declaration of `ModuleCacheEntry::get_data_raw()`, enforces a particular function signature for `Module::build_artifacts()`. I tried changing the closure on line 311, so that it didn't return whatever `SerializedModule::into_parts()`'s return value was directly by defining a mutable flag `enable_branch_protection` inside `Module::from_binary()`, but then the Rust compiler complained that it could not accept a closure that captured variables in that location.